tests/test_persistence.py

```diff
@@ -1,2200 +0,0 @@
-
-        def next_callback(update, context):
-            return NEXT2
-
-        next_handler = MessageHandler(None, next_callback)
-
-        def next2(update, context):
-            return ConversationHandler.END
-
-        next2 = MessageHandler(None, next2)
-
-        ch = ConversationHandler(
-            [start], {NEXT: [next_handler], NEXT2: [next2]}, [], name='name2', persistent=True
-        )
-        dp.add_handler(ch)
-        assert ch.conversations[ch._get_key(update)] == 1
-        dp.process_update(update)
-        assert ch._get_key(update) not in ch.conversations
-        update.message.text = '/start'
-        update.message.entities = [MessageEntity(MessageEntity.BOT_COMMAND, 0, 6)]
-        dp.process_update(update)
-        assert ch.conversations[ch._get_key(update)] == 0
-        assert ch.conversations == dict_persistence.conversations['name2']
-
-    def test_with_nested_conversationHandler(self, dp, update, conversations_json):
-        dict_persistence = DictPersistence(conversations_json=conversations_json)
-        dp.persistence = dict_persistence
-        NEXT2, NEXT3 = range(1, 3)
-
-        def start(update, context):
-            return NEXT2
-
-        start = CommandHandler('start', start)
-
-        def next_callback(update, context):
-            return NEXT2
-
-        next_handler = MessageHandler(None, next_callback)
-
-        def next2(update, context):
-            return ConversationHandler.END
-
-        next2 = MessageHandler(None, next2)
-
-        nested_ch = ConversationHandler(
-            [next_handler],
-            {NEXT2: [next2]},
-            [],
-            name='name3',
-            persistent=True,
-            map_to_parent={ConversationHandler.END: ConversationHandler.END},
-        )
-
-        ch = ConversationHandler(
-            [start], {NEXT2: [nested_ch], NEXT3: []}, [], name='name2', persistent=True
-        )
-        dp.add_handler(ch)
-        assert ch.conversations[ch._get_key(update)] == 1
-        assert nested_ch.conversations[nested_ch._get_key(update)] == 1
-        dp.process_update(update)
-        assert ch._get_key(update) not in ch.conversations
-        assert nested_ch._get_key(update) not in nested_ch.conversations
-        update.message.text = '/start'
-        update.message.entities = [MessageEntity(MessageEntity.BOT_COMMAND, 0, 6)]
-        dp.process_update(update)
-        assert ch.conversations[ch._get_key(update)] == 1
-        assert ch.conversations == dict_persistence.conversations['name2']
-        assert nested_ch._get_key(update) not in nested_ch.conversations
-        dp.process_update(update)
-        assert ch.conversations[ch._get_key(update)] == 1
-        assert ch.conversations == dict_persistence.conversations['name2']
-        assert nested_ch.conversations[nested_ch._get_key(update)] == 1
-        assert nested_ch.conversations == dict_persistence.conversations['name3']
-
-    def test_with_job(self, job_queue, dp):
```

Comment:
`test_{dict, pickle}_persistence` now exclusively test the interface of `Dict/PicklePersistence`. Testing the integration of persistence into the app is done exclusively in `test_basepersistence`. My intention is that on future changes on how the app interacts with the persistence, it will be more clear where tests are expected to fail/to be changed.